### PR TITLE
refactor: optimize pausable

### PR DESCRIPTION
**Pull request by 0xClandestine**

### CODE REVIEW

**Review by ypatil12 — COMMENTED** [review]
PR looks correct. How much are we saving storage-wise?

So the breaking change here is that we cannot reset the PauserRegistry without an upgrade. This _seems_ fine to me at first glance, will think more

**Review by ypatil12 — APPROVED** [review]

File: src/contracts/permissions/Pausable.sol [hunk 1]
```diff
@@ -20,114 +19,105 @@ import "../interfaces/IPausable.sol";
...
-        require(address(pauserRegistry) == address(0) && address(_pauserRegistry) != address(0), InputAddressZero());
-        _paused = initPausedStatus;
-        emit Paused(msg.sender, initPausedStatus);
-        _setPauserRegistry(_pauserRegistry);
+    /// Construction
+
+    constructor(
+        IPauserRegistry _pauserRegistry
+    ) {
+        pauserRegistry = _pauserRegistry;
```
Comment (wadealexc) [inline src/contracts/permissions/Pausable.sol:73]: Let's add a check that this is nonzero

File: src/contracts/permissions/Pausable.sol [hunk 2]
```diff
@@ -20,114 +19,105 @@ import "../interfaces/IPausable.sol";
...
+    /// @dev Returns a bitmap representing the paused status of the contract.
     uint256 private _paused;
 
-    uint256 internal constant UNPAUSE_ALL = 0;
-    uint256 internal constant PAUSE_ALL = type(uint256).max;
+    /// Modifiers
 
-    /// @notice
+    /// @dev Thrown if the caller is not a valid pauser according to the pauser registry.
     modifier onlyPauser() {
```
Comment (8sunyuan) [inline src/contracts/permissions/Pausable.sol:43]: Should we make `onlyPauser` and `onlyUnpauser` internal instead to save a little bit more gas?
`onlyPauser` is used twice in this contract but not something we ever use in inheriting contracts in core so the readability of having it as a modifier not as important imo.

**Review by 8sunyuan — APPROVED** [review]

File: src/contracts/permissions/Pausable.sol (hunk 2 above)
Comment (wadealexc) [reply src/contracts/permissions/Pausable.sol:43]: i checked, these don't save us bytecode

**Pull request merged**